### PR TITLE
Update nep-0508.md

### PR DESCRIPTION
**Pull request by shreyan-gupta**
Creating a PR so that we can discuss changes...

### CODE REVIEW

File: neps/nep-0508.md [hunk 1]
```diff
@@ -22,99 +22,92 @@ While the new approach addresses critical limitations left unsolved in NEP-40 an
...
 ### High level assumptions
 
-* Flat state is enabled.
-* Shard split boundary is predetermined. In other words, necessity of shard splitting is manually decided.
+* Flat storage is enabled.
+* Shard split boundary is predetermined and hardcoded. In other words, necessity of shard splitting is manually decided.
+* For the time being resharding as an event is only going to happen once but we would still like to have the infrastrcture in place to handle future resharding events with ease.
 * Merkle Patricia Trie is the undelying data structure for the protocol state.
-* Minimal epoch gap between two resharding events is X.
+* Epoch is atleast 6 hrs long for resharding to complete.
```
Comment (wacban) [inline neps/nep-0508.md:35]: nit: at least

File: neps/nep-0508.md [hunk 2]
```diff
@@ -22,99 +22,92 @@ While the new approach addresses critical limitations left unsolved in NEP-40 an
...
 * Resharding should work even when validators stop tracking all shards.
 * Resharding should work after stateless validation is enabled.
 * Resharding should be fast enough so that both state sync and resharding can happen within one epoch.
-* ~~Resharding should not require additional hardware from nodes.~~
-  * This needs to be assessed during test
-* Resharding should be fault tolerant
-  * A validator should be able to recover in case they go offline during resharding.
-    * For now, our aim is at least allowing a validator to join back after resharding is finished. 
+* Resharding should work efficiently within the limits of the current hardware requirements for nodes.
+* Resharding should be fault tolerant. A validator should be able to recover in case they go offline during resharding.
```
Comment (wacban) [inline neps/nep-0508.md:43]: Can you add that recovery will be a manual process requiring attention from the node operator?

File: neps/nep-0508.md [hunk 3]
```diff
@@ -22,99 +22,92 @@ While the new approach addresses critical limitations left unsolved in NEP-40 an
...
+The first release of the resharding v2 will contain a new shard layout where shard 3 will be split into two smaller shards. We chose shard 3 for this purpose as that was the largest of the 4 shards. Furthermore additional reshardings can be scheduled in subsequent neard releases without additional NEPs unless the need for it arises. A new shard layout can be determined and will be scheduled and executed with the next protocol upgrade. Resharding will typically happen by splitting one of the existing shards into two smaller shards. The new shard layout will be created by adding a new boundary account that will be determined by analysing the storage and gas usage metrics within the shard and selecting a point that will divide the shard roughly in half in accordance to the mentioned metrics. Other metrics can also be used based on requirements.
```
Comment (wacban) [inline neps/nep-0508.md:99]: @walnut-the-cat suggested earlier to not specify which shard we'll be splitting in the NEP

**Review by wacban — APPROVED** [review]
looks great, thanks!

File: neps/nep-0508.md (hunk 3 above)
Comment (shreyan-gupta) [reply neps/nep-0508.md]: Reverted to original

File: neps/nep-0508.md (hunk 2 above)
Comment (shreyan-gupta) [reply neps/nep-0508.md]: Change the fault tolerant point here

File: neps/nep-0508.md [hunk 4]
```diff
@@ -106,116 +106,93 @@ This was implemented ahead of this NEP and the fixed shards feature was **remove
...
+* When syncing the state for the first epoch where the new shard layout is used. In this case the node would need to apply the last block of the previous epoch. It cannot be done on the children shard as on chain the block was applied on the parent shards and the trie related gas costs would be different.
 * When generating proofs for incoming receipts. The proof for each of the children shards contains only the receipts of the shard but it's generated on the parent shard layout and so may not be verified. 
 
-In this NEP we propose that resharding should be rolled out first, before any real dependency on state sync is added. We can then safely roll out the resharding logic and solve the abovementioned issues separately. 
+In this NEP we propose that resharding should be rolled out first, before any real dependency on state sync is added. We can then safely roll out the resharding logic and solve the above mentioned issues separately. We believe atleast some of the issues can be mitigated by the new implementation of post-state root and chunk execution design.
```
Comment (wacban) [inline neps/nep-0508.md:143]: It's not longer post state root, it's "new pre state root" :D

File: neps/nep-0508.md [hunk 5]
```diff
@@ -106,116 +106,93 @@ This was implemented ahead of this NEP and the fixed shards feature was **remove
...
 * Underlying trie structure and data structure are not going to change.
-* Resharding will create dependency on flat state snapshots. 
+* Resharding will create dependency on flat state snapshots.
 
 ### Negative
 
-* The resharding process is still not fully automated. Analyzing shard data, determining the split boundary, and triggering an actual shard split all need to be manually curated by a person.
-* During resharding, a node is expected to do more work as it will first need to copy a lot of data around the then will have to apply changes twice (for the current shard and the future shard).
+* The resharding process, as of now, is not fully automated. Analyzing shard data, determining the split boundary, and triggering an actual shard split all need to be manually curated and tracked.
+* During resharding, a node is expected to require more resources as it will first need to copy state data from the parent shard to the child shard, the then will have to apply trie and flat state changes twice, once for the parent shard and once for the child shards.
```
Comment (wacban) [inline neps/nep-0508.md:189]: nit: the then

File: neps/nep-0508.md [hunk 6]
```diff
@@ -106,116 +106,93 @@ This was implemented ahead of this NEP and the fixed shards feature was **remove
...
-
-* What parts of the design do you expect to resolve through the NEP process before this gets merged?
-* What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
-* What related issues do you consider out of scope for this NEP that could be addressed in the future independently of the solution that comes out of this NEP?]
-
-## Changelog
-
-[The changelog section provides historical context for how the NEP developed over time. Initial NEP submission should start with version 1.0.0, and all subsequent NEP extensions must follow [Semantic Versioning](https://semver.org/). Every version should have the benefits and concerns raised during the review. The author does not need to fill out this section for the initial draft. Instead, the assigned reviewers (Subject Matter Experts) should create the first version during the first technical review. After the final public call, the author should then finalize the last version of the decision context.]
-
-### 1.0.0 - Initial Version
```
Comment (wacban) [inline neps/nep-0508.md:205]: I thought this should stay here and be filled in when reviewed. No clue if that is how it works actually.

**Review by wacban — APPROVED** [review]
LGTM!

File: neps/nep-0508.md (hunk 4 above)
Comment (shreyan-gupta) [reply neps/nep-0508.md:143]: What? We went back to pre state root?

File: neps/nep-0508.md (hunk 6 above)
Comment (shreyan-gupta) [reply neps/nep-0508.md:205]: Oh right, sorry, I just took a look at some other NEPs and they didn't have this, so maybe I thought it shouldn't be there. My bad, will add in

**Pull request merged**